File: backends/Contracts/Common.Infra/Cache/Redis/Handlers/Remove.cs
using D2.Contracts.Common;
using D2.Contracts.Common.App;
using H = D2.Contracts.Common.App.ICacheService.IRemoveHandler;
using I = D2.Contracts.Common.App.ICacheService.RemoveInput;
using O = D2.Contracts.Common.App.ICacheService.RemoveOutput;

namespace Common.Infra.Cache.Redis.Handlers;

/// <inheritdoc cref="H"/>
public class Remove : BaseHandler<H, I, O>, H
{
    /// <summary>
    /// Initializes a new instance of the <see cref="Remove"/> class.
    /// </summary>
    ///
    /// <inheritdoc/>
    public Remove(IHandlerContext context) : base(context) { }

    /// <inheritdoc/>
    protected override async ValueTask<D2Result<O?>> ExecuteAsync(
        I input,
        CancellationToken ct = default)
    {
        // TODO: Implement.
        return default;
    }
}

File: backends/node/shared/interfaces/src/caching/in-memory/handlers/d/remove.ts
import type { IHandler } from "@d2/handler";

/** Input for removing a single key from the in-memory cache. */
export interface RemoveInput {
  key: string;
}

/** Output for removing a single key from the in-memory cache. */

export interface RemoveOutput {}

/** Handler for removing a single key from the in-memory cache. */
export type IRemoveHandler = IHandler<RemoveInput, RemoveOutput>;
